### PR TITLE
Refs #24121 -- Added __repr__() to AdminSite, DefaultAdminSite, and ModelAdmin.

### PR DESCRIPTION
**Pull request by Edald123**

### CODE REVIEW

**Comment by felixxm** [comment]
@Edald123 Thanks :+1: Please fix `flake8` and `isort` errors.

File: tests/admin_default_site/tests.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from django.contrib import admin
 from django.contrib.admin import sites
 from django.test import SimpleTestCase, override_settings
+from .sites import CustomAdminSite
```
Comment (felixxm) [inline tests/admin_default_site/tests.py:4]: Blank line is missing:
```suggestion

from .sites import CustomAdminSite
```

File: django/contrib/admin/options.py [hunk 2]
```diff
@@ -594,6 +594,12 @@ def __init__(self, model, admin_site):
     def __str__(self):
         return "%s.%s" % (self.model._meta.app_label, self.__class__.__name__)
 
+    def __repr__(self):
+        return (
+            f'<{self.__class__.__qualname__}: model={self.model.__qualname__} '
+            f'site={self.admin_site!r}>'        
+        )
```
Comment (felixxm) [inline django/contrib/admin/options.py:601]:
```suggestion
            f'site={self.admin_site!r}>'
        )
```

File: tests/modeladmin/tests.py [hunk 3]
```diff
@@ -722,6 +722,12 @@ def has_delete_permission(self, request, obj=None):
         self.assertEqual(perms_needed, {'band'})
         self.assertEqual(protected, [])
 
+    def test_modeladmin_repr(self):
+        ma = ModelAdmin(Band, self.site)
+        self.assertEqual(
+            repr(ma),
+            "<ModelAdmin: model=Band site=AdminSite(name='admin')>",
+        )
```
Comment (felixxm) [inline tests/modeladmin/tests.py:730]:
```suggestion
        )

```

**Comment by Edald123** [comment]
I will fix the in a minute, thanks!

**Comment by felixxm** [comment]
@Edald123 Thanks :+1: Welcome aboard :boat:

**Pull request merged**